### DISCUSSION TITLE
How does one update Debian OS (apt) packages in a CT/LXC? Do we even need to?

### DISCUSSION THREAD

**Discussion by alichaudry** [tteck/Proxmox#2871] [Q&A]
Hello, I recently installed the AdGuard Home (AGH) CT using the script in this repo. This is my first time using CTs (LXC containers) in Proxmox so please excuse my ignorance if these are particularly dumb questions; I saw nothing regarding these questions on the site for this repo (as well as the various readmes/guides inside the repo), and I did try searching for some key terms (`apt`, `update lxc`, etc.) in the Issues and Discussions sections of the repo as well to no avail. Questions as follows:

1. How do I upgrade OS packages in the container? I see `apt` still works and it generally feels like a proper VM (other than the fact that the kernel is shared w/ the PVE host), so maybe just an `apt-get update && apt-get upgrade` should suffice?
2. Do I even _need_ to update system packages? I imagine there's always the risk of incompatibilities arising with the software (AGH in this case) that could be avoided by just not updating the CT.
3. There's also the risk of not patching vulnerabilities if no OS upgrades are applied, so maybe I should turn on unattended security upgrades?
4. I skimmed the container update script, and I don't see any references to `apt update` or `apt-get update` in there, including in the script it sources at the very top of the update file. Does the container update script only update the underlying software (AGH) and _not_ the `apt` packages?

Also happy to research this myself if I were to be pointed in the right direction. Thank you.

**Comment by rfgamaral** [comment]
Isn't [this](https://helper-scripts.com/scripts?id=Proxmox+VE+LXC+Updater) what you're looking for?

**Comment by Mellowlynx** [comment]
1. This depends on the OS, some use apt, some apk, others something else.
2. I would, but be sure to back up before you run the update (on OS or app), so you can restore is something breaks.
3. Up to you, I run an update playbook trough Ansible twice a week, it runs a view hours after Proxmox made a backup.
4. It's there, just `apt-get update` is used, not `apt update`. Lines 67–70 show the different OS options and what is executed.
